### PR TITLE
11 crud student pull request

### PR DESCRIPTION
**Pull request by jvcanavarro**

### CODE REVIEW

**Comment by renan-cunha** [comment]
You should resolve some conflicts

File: mobile/lib/core/data/database.dart [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import 'package:mobile/features/class_management/data/models/classroom_model.dart';
 import 'package:mobile/features/student_management/data/models/student_model.dart';
+import 'package:mobile/features/student_management/domain/use_cases/delete_student_use_case.dart';
```
Comment (renan-cunha) [inline mobile/lib/core/data/database.dart:3]: You don't need use_cases in the database file

File: mobile/lib/features/class_management/data/repositories/classroom_repository_impl.dart [hunk 2]
```diff
@@ -25,27 +24,27 @@ class ClassroomRepositoryImpl implements ClassroomRepository {
     return await _tryCacheClassroom(classroom);
   }
 
-  @override
-  Future<Either<Failure, void>> deleteClassroom(Classroom classroom) async {
-    return await _tryDeleteClassroom(classroom);
-  }
-
-  Future<Either<Failure, void>> _tryDeleteClassroom(Classroom classroom) async {
+  Future<Either<Failure, Classroom>> _tryCacheClassroom(
```
Comment (renan-cunha) [inline mobile/lib/features/class_management/data/repositories/classroom_repository_impl.dart:27]: changes in classroom feature, is this correct?

File: mobile/lib/features/student_management/data/data_sources/student_local_data_source.dart [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+  }
+
+  @override
+  Future<void> deleteStudentFromCache(StudentModel studentModel) async {
+    var pk = studentModel.localId;
+    try {
+      var done = await this.database.deleteStudent(pk);
+      if (done != 1) {
+        throw SqliteException(787, "The table don't have this entry");
```
Comment (renan-cunha) [inline mobile/lib/features/student_management/data/data_sources/student_local_data_source.dart:57]: Typo

**Review by renan-cunha — CHANGES_REQUESTED** [review]
I think you are missing the tests of the database manipulation with the student table

File: mobile/lib/features/student_management/data/data_sources/student_local_data_source.dart (hunk 3 above)
Comment (jvcanavarro) [reply mobile/lib/features/student_management/data/data_sources/student_local_data_source.dart:57]: don't get it

Comment (VitorCantao) [reply mobile/lib/features/student_management/data/data_sources/student_local_data_source.dart:57]: The table doesn't have this entry

**Comment by jvcanavarro** [comment]
WTF a just changed from **don't** to **doesn't** and udpate `database.dart` now coverate isn't passing

**Comment by renan-cunha** [comment]
> WTF a just changed from **don't** to **doesn't** and udpate `database.dart` now coverage isn't passing

It's because the develop branch was updated. You are missing the tests with the database

File: mobile/lib/features/student_management/data/data_sources/student_local_data_source.dart (hunk 3 above)
Comment (VitorCantao) [reply mobile/lib/features/student_management/data/data_sources/student_local_data_source.dart]: > WTF a just changed from **don't** to **doesn't** and udpate `database.dart` now coverage isn't passing

Probably the database.dart, with one new line

**Comment by jvcanavarro** [comment]
you sure? missing at least 1 test should not decread corevage by more than 1% at least?

**Comment by renan-cunha** [comment]
> you sure? missing at least 1 test should not decrease corevage by more than 1%?

not 1, but four tests since you added four functions

**Review by jvcanavarro — COMMENTED** [review]
> > you sure? missing at least 1 test should not decrease corevage by more than 1%?
> 
> not 1, but four tests since you added four functions

So the coverage must had reduced a lot instead of only 0,08%. Anyway I'm checking this now.

**Comment by renan-cunha** [comment]
> > > you sure? missing at least 1 test should not decrease corevage by more than 1%?
> > 
> > 
> > not 1, but four tests since you added four functions
> 
> So the coverage must had reduced a lot instead of only 0,08%. Anyway I'm checking this now.

this is the file that needs tests 
image

**Comment by jvcanavarro** [comment]
> > > you sure? missing at least 1 test should not decrease corevage by more than 1%?
> > 
> > 
> > not 1, but four tests since you added four functions
> 
> So the coverage must had reduced a lot instead of only 0,08%. Anyway I'm checking this now.



> > > > you sure? missing at least 1 test should not decrease corevage by more than 1%?
> > > 
> > > 
> > > not 1, but four tests since you added four functions
> > 
> > 
> > So the coverage must had reduced a lot instead of only 0,08%. Anyway I'm checking this now.
> 
> this is the file that needs tests
> image

no no thanks

**Comment by jvcanavarro** [comment]
@renan-cunha review pls

**Review by renan-cunha — APPROVED** [review]
Hell yeah

**Pull request merged**